packages/flutter_tools/test/general.shard/dart/language_version_test.dart

```diff
@@ -0,0 +1,228 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:file/memory.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/dart/language_version.dart';
+import 'package:package_config/package_config.dart';
+
+import '../../src/common.dart';
+
+void main() {
+  testWithoutContext('detects language version in comment', () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final File file = fileSystem.file('example.dart')
+      ..writeAsStringSync('''
+// Some license
+
+// @dart = 2.9
+''');
+
+    expect(determineLanguageVersion(file, null), '// @dart = 2.9');
+  });
+
+  testWithoutContext('detects technically invalid language version', () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final File file = fileSystem.file('example.dart')
+      ..writeAsStringSync('''
+// Some license
+
+// @dart
+''');
+
+    expect(determineLanguageVersion(file, null), '// @dart');
+  });
+
+  testWithoutContext('detects language version with leading whitespace', () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final File file = fileSystem.file('example.dart')
+      ..writeAsStringSync('''
+// Some license
+
+    // @dart = 2.9
+''');
+
+    expect(determineLanguageVersion(file, null), '// @dart = 2.9');
+  });
+
+  testWithoutContext('detects language version with tabs', () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final File file = fileSystem.file('example.dart')
+      ..writeAsStringSync('''
+// Some license
+
+//\t@dart = 2.9
+''');
+
+    expect(determineLanguageVersion(file, null), '//\t@dart = 2.9');
+  });
+
+  testWithoutContext('detects language version with tons of whitespace', () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final File file = fileSystem.file('example.dart')
+      ..writeAsStringSync('''
+// Some license
+
+//        @dart       = 23
+''');
+
+    expect(determineLanguageVersion(file, null), '//        @dart       = 23');
+  });
+
+  testWithoutContext('does not detect language version in dartdoc', () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final File file = fileSystem.file('example.dart')
+      ..writeAsStringSync('''
+// Some license
+
+/// @dart = 2.9
+''');
+
+    expect(determineLanguageVersion(file, null), null);
+  });
+
+  testWithoutContext('does not detect language version in block comment', () {
+    final FileSystem fileSystem = MemoryFileSystem.test();
+    final File file = fileSystem.file('example.dart')
+      ..writeAsStringSync('''
+// Some license
+
+/*
+@dart = 2.9
+*/
```

Comment:
Missing the // prefix so i's not really detecting it's in a block `//@dart = 2.9`

Reply:
Fixed